### PR TITLE
Use product-details instead of the wiki

### PR DESCRIPTION
**Pull request by sylvestre**
auto nag will still fail if the wiki is down because of 
cycle_span = getTemplateValue("https://wiki.mozilla.org/Template:CURRENT_CYCLE")
but this is the first step

### CODE REVIEW

**Comment by jcristau** [comment]
Care to expand the commit message to explain why product-details is better than the wiki?

File: auto_nag/scripts/query_creator.py [hunk 1]
```diff
@@ -9,6 +9,13 @@
 from auto_nag.bugzilla.utils import (get_config_path, get_project_root_path,
                                      createQueriesList, cleanUp)
 
+try:
+    # For Python 3.0 and later
+    from urllib.request import urlopen
+except ImportError:
+    # Fall back to Python 2's urllib2
+    from urllib2 import urlopen
+
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: relman-auto-nag is nowhere near py3k-ready.  I'm not sure adding compat code in one place adds more value than confusion, unless there's a bigger effort to convert it?  (Same with the print below)

File: auto_nag/scripts/query_creator.py [hunk 2]
```diff
@@ -17,16 +24,29 @@ def getTemplateValue(url):
...
 
+def loadJSON(url):
+    jsonurl = urlopen(url)
+    return json.loads(jsonurl.read())
+
+
+def getVersion(json, key):
+    # In X.Y, we just need X
+    version = json[key].split(".")
+    return version[0]
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: overloading the json global, use a different variable name?

File: auto_nag/scripts/query_creator.py [hunk 3]
```diff
@@ -99,7 +113,7 @@ def getReportURL(approval_flag, span):
     queries = createQueriesList(queries_dir, weekday, urls, print_all=options.queries_only)
     if options.queries_only:
         for url in urls:
-            print url
+            print(url)
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: drop this hunk

File: auto_nag/scripts/query_creator.py [hunk 4]
```diff
@@ -17,16 +18,29 @@ def getTemplateValue(url):
...
 no_nag = ";field3-1-0=status_whiteboard;type3-1-0=notsubstring;value3-1-0=[no-nag]"
-beta_version = getTemplateValue("https://wiki.mozilla.org/Template:BETA_VERSION")
-aurora_version = getTemplateValue("https://wiki.mozilla.org/Template:AURORA_VERSION")
-central_version = getTemplateValue("https://wiki.mozilla.org/Template:CENTRAL_VERSION")
-esr_version = getTemplateValue("https://wiki.mozilla.org/Template:ESR_VERSION")
+jsonContent = loadJSON("https://product-details.mozilla.org/1.0/firefox_versions.json")
+beta_version = getVersion(jsonContent, "LATEST_FIREFOX_DEVEL_VERSION")
+aurora_version = getVersion(jsonContent, "FIREFOX_AURORA")
+central_version = getVersion(jsonContent, "FIREFOX_NIGHTLY")
+esr_version = getVersion(jsonContent, "FIREFOX_ESR")
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: this reverts us from esr52 to esr45 for the next few weeks, which seems like a bad thing (45 is essentially dead for non-chemspilly things now).  can we hold off on this patch until 45 is no more, or look at ESR_NEXT, or something  else?

**Comment by sylvestre** [comment]
I updated the commit message with more information.

File: auto_nag/scripts/query_creator.py [hunk 5]
```diff
@@ -17,16 +18,36 @@ def getTemplateValue(url):
     return parsed_template.groups()[0]
 
 
+def loadJSON(url):
+    jsonurl = urlopen(url)
+    return json.loads(jsonurl.read())
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: return json.load(urlopen(url))

File: auto_nag/scripts/query_creator.py [hunk 6]
```diff
@@ -17,16 +18,36 @@ def getTemplateValue(url):
...
-beta_version = getTemplateValue("https://wiki.mozilla.org/Template:BETA_VERSION")
-aurora_version = getTemplateValue("https://wiki.mozilla.org/Template:AURORA_VERSION")
-central_version = getTemplateValue("https://wiki.mozilla.org/Template:CENTRAL_VERSION")
-esr_version = getTemplateValue("https://wiki.mozilla.org/Template:ESR_VERSION")
+jsonContent = loadJSON("https://product-details.mozilla.org/1.0/firefox_versions.json")
+beta_version = getVersion(jsonContent, "LATEST_FIREFOX_DEVEL_VERSION")
+aurora_version = getVersion(jsonContent, "FIREFOX_AURORA")
+central_version = getVersion(jsonContent, "FIREFOX_NIGHTLY")
+esr_next_version = getVersion(jsonContent, "FIREFOX_ESR_NEXT")
+if not len(esr_next_version):
```
Comment (jcristau) [inline auto_nag/scripts/query_creator.py]: this seems to leave esr_version unset if there *is* a ESR_NEXT.  also no need to call len() explicitly.

**Pull request merged**